paper/paper.tex

```diff
@@ -0,0 +1,946 @@
+
+There are several benefits to working in log-space. Results are much less likely to overflow or underflow, and the many products and exponents become sums and products, respectively. which are much more efficient to compute and to differentiate.
+
+Working locally and in log-space also gives us a convenient \emph{antisymmetry},
+\begin{verbatim}
+logdensity_rel(μ, ν, x) == -logdensity_rel(ν, μ, x)
+\end{verbatim}
+
+In particular, \verb|logdensity_rel(μ, ν, x)| takes special values in some common cases:
+\newcommand{\notllx}{\mathrel{\centernot{\ll}\!\!_x}}
+
+% works now
+\begin{center}
+\begin{tabular}{@{}rrr@{}}\toprule
+%& \multicolumn{2}{c}{$w = 8$}\\\cmidrule{2-3} 
+& $\nu\ll_x \mu$& \text{(else)} \\\midrule
+$\mu\ll_x \nu$& (finite) & \verb|-Inf|  \\
+\text{(else)} & \verb|Inf| & \verb|NaN|  \\\bottomrule
+\end{tabular}
+\end{center}
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+\section{Base measures and log-densities}
+It's very useful for users to be able to \emph{call} \verb|logdensity_rel| for arbitrary measures. But \emph{defining} things directly in these terms would require a definition for every pair of possible measures, which is of course intractable.
+
+We'll now discuss the mechanics of defining a measure, followed by a description of the end-user perspective.
+
+\subsection*{Defining a measure}
+
+To define a new measure requires a \emph{base measure}, the \emph{log-density} with respect to that base measure, and a \emph{support}. 
+A base measure can be defined locally or globally. In the latter case, the local base measure defaults to 
+\begin{verbatim}
+basemeasure(μ::AbstractMeasure, x) = basemeasure(μ)
+\end{verbatim}
+This allows users to define a base measure specific to a given neighborhood if they prefer, or to define a global base measure when that suffices. 
+
+For example, for a standard \verb|Normal| measure, we have
+
+\begin{verbatim}
+basemeasure(::Normal{()}) = (1/sqrt2π) * Lebesgue(ℝ)
+
+logdensity_def(::Normal{()} , x) = - x^2 / 2 
+
+insupport(::Normal{()}, x) = true
+\end{verbatim}
+
+Note the difference from a typical distribution-oriented implementation; the log-density consists only of $-x^2/2$, with the normalization term pushed into the base measure. 
+
+The normalization term is present for any normal distribution, even those with a different mean or variance. So if we have a product of normals, for example in a regression problem, the normalization can be shared across them. For the simple example of $n$ iid standard \verb|Normal|s, this effectively rewrites the log-pdf
+\[
+    \sum_{j=1}^n \left(-\frac{1}{2}\log 2π -\frac{x_j^2}{2}\right)
+    \quad \text{as} \quad
+    -\frac{n}{2}\log 2π + \sum_{j=1}^n \left(-\frac{x_j^2}{2}\right)
+\]
+
+Further simplification is clearly possible; doing this in a composable way without sacrificing efficiency is the subject of ongoing work.
+
+\subsection*{The end-user perspective}
```

Comment:
Needs a warning that for inference it is important if parameters are in the base measure - dominated experiment and a way to query that

Reply:
Can you give some more detail? Not sure I understand

Reply:
If N(\mu, \sigma) has a base measure N(0, \sigma), this is good when estimating mu, but tricky when estimating $sigma$

Reply:
Ah right, yes this comes up a lot. If we have two measures that depend on some variable and want to compare them, we need to use a base measure that's invariant wrt that variable. It's kind of like in physics how you want an inertial frame of reference.

Reply:
Just write that!